src/cincinnati/mod.rs

```diff
@@ -55,6 +56,7 @@ lazy_static::lazy_static! {
         "Total number of errors while checking for updates.",
         &["kind"]
     ).unwrap();
+    static ref DEADEND_RELEASE_EXISTS : AtomicBool = AtomicBool::new(false);
```

Comment:
As I mentioned in my [previous review](https://github.com/coreos/zincati/pull/367#discussion_r498668206), this is a [tristate](https://en.wikipedia.org/wiki/Three-valued_logic) (`unknown` / `true` / `false`), not a boolean.

In particular, when the service starts it doesn't know its deadend state (i.e. it starts in `unknown` mode) until the first fetch. This currently assumes the service is always starting from `false`, which is not guaranteed if the service was running before with a `true` state and then got restarted.

You can model this by wrapping an [AtomicU8](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU8.html) into a new object and exposing relevant set and query methods.